### PR TITLE
Add UI for cancellation of jobs

### PR DESCRIPTION
**Pull request by j16r**
e.g:

`tools/bin/cldev runs cancel 42fda7a0ff7041cdad58401d28971b44`

Should now cancel a running or pending job.

[Finished #168117690]

### CODE REVIEW

File: core/cmd/client.go [hunk 1]
```diff
@@ -167,6 +168,11 @@ func (h *authenticatedHTTPClient) Post(path string, body io.Reader) (*http.Respo
 	return h.doRequest("POST", path, body)
 }
 
+// Put performs an HTTP Post using the authenticated HTTP client's cookie.
```
Comment (rupurt) [inline core/cmd/client.go]: `HTTP Post` -> `HTTP Put`?

File: core/cmd/remote_client.go [hunk 2]
```diff
@@ -630,3 +630,20 @@ func (cli *Client) GetConfiguration(c *clipkg.Context) error {
 	cwl := presenters.ConfigWhitelist{}
 	return cli.renderAPIResponse(resp, &cwl)
 }
+
+// CancelJob soft deletes a job and its associated runs.
```
Comment (rupurt) [inline core/cmd/remote_client.go]: Is this comment copy + pasta?

File: core/cmd/remote_client_test.go [hunk 3]
```diff
@@ -879,3 +879,34 @@ func TestClient_GetConfiguration(t *testing.T) {
 	assert.Equal(t, cwl.Whitelist.RootDir, app.Config.RootDir())
 	assert.Equal(t, cwl.Whitelist.SessionTimeout, app.Config.SessionTimeout())
 }
+
+func TestClient_CancelJob(t *testing.T) {
```
Comment (rupurt) [inline core/cmd/remote_client_test.go]: `TestClient_CancelJob` -> `TestClient_CancelJobRun`

File: core/web/job_runs_controller.go [hunk 4]
```diff
@@ -143,3 +143,19 @@ func (jrc *JobRunsController) Update(c *gin.Context) {
 		jsonAPIResponse(c, jr, "job run")
 	}
 }
+
+// Cancel stops a Run from continuing.
+// Example:
+//  "<application>/runs/:RunID/cancellation"
+func (jrc *JobRunsController) Cancel(c *gin.Context) {
```
Comment (se3000) [inline core/web/job_runs_controller.go]: Can we get a test for this?

Comment (j16r) [reply core/web/job_runs_controller.go]: Good catch, added.

**Pull request merged**